### DISCUSSION TITLE
2022-01-17 weekly planning

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#979] [Maintainer discussions]
## About
At the start of each week, we start a discussion covering Mathesar's progress, team member check ins, and general updates. Everyone is welcome to add a new topic to discuss or to comment on existing topics.

## Discussion topics
- [Frontend update](https://github.com/centerofci/mathesar/discussions/979#discussioncomment-1972735)
- [Backend update](https://github.com/centerofci/mathesar/discussions/979#discussioncomment-1972736)
- [Design update](https://github.com/centerofci/mathesar/discussions/979#discussioncomment-1972738)
- [Product update](https://github.com/centerofci/mathesar/discussions/979#discussioncomment-1972742)
- [Comms Assignee schedule](https://github.com/centerofci/mathesar/discussions/979#discussioncomment-1972744)
- [Dominykas' check in](https://github.com/centerofci/mathesar/discussions/979#discussioncomment-1984586)
- [Brent's check in](https://github.com/centerofci/mathesar/discussions/979#discussioncomment-1985201)
- [Pavish's check in](https://github.com/centerofci/mathesar/discussions/979#discussioncomment-1985359)
- [Ghislaine's check-in](https://github.com/centerofci/mathesar/discussions/979#discussioncomment-1985669)
- [Mukesh's check-in](https://github.com/centerofci/mathesar/discussions/979#discussioncomment-1990509)
- [Sean's check-in](https://github.com/centerofci/mathesar/discussions/979#discussioncomment-1990757)
- [Kriti's check in](https://github.com/centerofci/mathesar/discussions/979#discussioncomment-1992729)

## Weekly check-in
@centerofci/mathesar-core please post your weekly check-in as a top-level response and edit the discussion topics above to add your check-in as a topic. Others are also welcome to respond.

Prompts:
- Any thoughts on the concept/format of the new Views product spec?
- How did your most recent work go?
- What are you planning to work on soon?
- Do you anticipate any dependencies or blockers that someone else on the team can help you with?

If you'd like to add anything else to your check-in that's not related to the prompts, feel free to do so.

**Comment by kgodey** [comment]
## Frontend update

### Completed
1. PRs merged:
	1. #915
	2. #975
2. Progress made on:
	1. #826
	2. #861
	3. #870
	4. #926
	5. #961
	6. #976
	7. #977

### Current Priorities
1. Review and merge PRs ready for review. 
	1. #826
	2. #861
	3. #961
	4. #976
	5. #977
2. Get #926 ready to review
3. Continue to make progress on the [Working with Tables](https://github.com/centerofci/mathesar/milestone/8) milestone. There only seem to be two unblocked issues.
    1. #500 
    2. #531
4. Work on some of the [14 unassigned frontend issues](https://github.com/centerofci/mathesar/issues?q=is%3Aopen+is%3Aissue+label%3A%22work%3A+frontend%22+milestone%3A%22[07.1]+2022-01+improvements%22+no%3Aassignee) in the [2022-01 improvements milestone](https://github.com/centerofci/mathesar/milestone/33) while we're waiting for the type-related frontend PRs to be merged and more backend work to be done on data types. Feel free to also move some into later milestones if they're not critical.

@pavish @seancolsen Please comment below with thoughts/concerns.

**Reply by pavish** [reply]
Sounds good to me

**Reply by seancolsen** [reply]
Looks good

**Comment by kgodey** [comment]
## Backend update

### Completed

1. Merged PRs:
	1. #885
	2. #965
	3. #966
	4. #969
2. Progress has been made on the following PRs:
    1. #909
    2. #921

### Current priorities

1. API error handling issues:
    1.  #859 (top priority since it's blocking the frontend)
    2. #623
    3. #883 
    4. #889
2. Finalize the approach to filtering (please prioritize responding to @dmos62's requests promptly) and get these draft PRs closer to merge:
    1. #921 
    2. #879 
    3. #893
3. Continue to make progress on other [Initial Data Types](https://github.com/centerofci/mathesar/milestone/11)  issues. Some good candidates might be:
    1. Data Types:
        1. #430 (assigned to @silentninja)
        2. #387 (assigned to @mathemancer)
        3. #420 (assigned to @mathemancer)
        4. #417 (assigned to @mathemancer)
    2. Date & Time parsing: #845
    3. Grouping:
        1. #408 
        2. #411 
        3. #414 
        4. #384 (assigned to @mathemancer)
        5. #429

@dmos62 @mathemancer @silentninja Please comment below with thoughts/concerns. If any of you have too many issues assigned, please unassign yourself from any you haven't started yet so someone else can take them.

**Reply by mathemancer** [reply]
Looks reasonable.  Keep in mind that the filtering and grouping work may interfere with each other.

**Reply by silentninja** [reply]
Looks good. I have unassigned myself from [column name -> id](https://github.com/centerofci/mathesar/issues/839) issue, will be working on it once the error handling PR is ready for review

**Reply by kgodey** [reply]
@mathemancer If you don't want to get started on the grouping work until filtering is resolved, I recommend doing the data types instead. Since @silentninja has a lot of issues in progress already, perhaps you could take over `INTERVAL` from him?

**Reply by mathemancer** [reply]
Sure, I'll jump into that.

**Comment by kgodey** [comment]
## Design update

### Completed
1.  https://github.com/centerofci/mathesar-wiki/pull/16 is now merged
2. Review almost complete on spec: https://github.com/centerofci/mathesar-wiki/pull/19
3. Review almost complete on spec updates for #960
4. Exploratory work started for #466

### Current priorities 
1. Get currently open specs reviewed and hopefully merged:
    1. https://github.com/centerofci/mathesar-wiki/pull/19
    2. https://github.com/centerofci/mathesar-wiki/pull/21
2. Spec for #978 
3. https://github.com/centerofci/mathesar-wiki/issues/18
4. Exploratory work for Views in the design prototype. We're holding off on any more Views specs until https://github.com/centerofci/mathesar-wiki/pull/22 is through review.

@ghislaineguerin @ppii775 Please comment below with thoughts/concerns

**Reply by ghislaineguerin** [reply]
No concerns. @ppii775 feel free to reach out with questions.

**Reply by ppii775** [reply]
@ghislaineguerin Thanks! No questions for now – I'll dive into lists more. The new semester has started and I can work on Mathesar mostly on Fridays, FYI.

**Reply by kgodey** [reply]
Sounds good, @ppii775!

**Comment by kgodey** [comment]
## Product update

### Completed
1. First draft of https://github.com/centerofci/mathesar-wiki/pull/22
2. Almost new all tickets are finalized for the following specs:
    1. https://github.com/centerofci/mathesar-wiki/pull/12. 
    2. https://github.com/centerofci/mathesar-wiki/pull/13
3. Fleshed out #978

### Current priorities
1. Get https://github.com/centerofci/mathesar-wiki/pull/22 reviewed and merged.
2. Figure out remaining Views product requirements based on spec feedback and known holes (e.g. JOINs)
3. Finalize https://github.com/centerofci/mathesar/issues/944
4. Think through representing filtering based on CTEs to users (and when that would make sense in our roadmap), see:
    1. #846 
    2. #853 
5. Create tickets for spec: https://github.com/centerofci/mathesar-wiki/pull/16

**Comment by kgodey** [comment]
## Comms Assignee schedule

| Week | Person |
|-|-|
| 2022-01-17 to 2022-01-21 | Kriti |
| 2022-01-24 to 2022-01-28 | Pavish |
| 2022-01-31 to 2022-02-04 | Mukesh |
| 2022-02-07 to 2022-02-11 | Sean |
| 2022-02-14 to 2022-02-18 | Dominykas |
| 2022-02-21 to 2022-02-25 | Ghislaine |
| 2022-02-28 to 2022-03-04 | Brent |

### Instructions
You can also find the schedule and instructions for the Comms Assignee on the [wiki page](https://wiki.mathesar.org/team/guide/comms-assignee). Please feel free to update the wiki page if you think of anything useful during your shift.

**Comment by dmos62** [comment]
> Any thoughts on the concept/format of the new Views product spec?

I have been predominantly focused on the filtering refactor and have not been tracking this spec's development as closely as I'd like. That said, I'm pretty happy with the current state of the views product spec.

> How did your most recent work go?

Suggested a more generalized approach to how we approach filtering. Our iterations seemed to be pointing that way. Went well.

> What are you planning to work on soon?

Continuing feedback-iteration loop on the filtering refactor.

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

I'll be waiting on feedback from time to time. I'm not worried, since feedback last week was very quick, and I've a backlog of tasks for when I'm blocked.

**Comment by mathemancer** [comment]
## Brent's check-in

> Any thoughts on the concept/format of the new Views product spec?

I think the format works well.  I'll add more details in that spec discussion.

> How did your most recent work go?

Pretty well.  I got a few smallish PRs in, and merged some PRs for others.  Overall, I think the smaller issue strategy is working for me at the moment.

> What are you planning to work on soon?

Anticipating filtering being complete (or at least the major structural changes being outlined) soon, I'm getting back into grouping.  For the moment, just setting up some SQL utils that will do some calculations to define numeric ranges.  I'll also work on the money type(s).

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

Just the oft-mentioned dependency between filtering and grouping.  Will coordinate with @dmos62 on it.

**Comment by pavish** [comment]
## Pavish's check-in

> Any thoughts on the concept/format of the new Views product spec?

I think it's pretty good. I'll add more thoughts on the discussion.

> How did your most recent work go?

The form builder PR is merged and I reviewed some open PRs. I had some ambitious goals last week inorder to get the data types milestone on the frontend ready by the end of Jan, but didn't manage to meet most of them. Work is progressing though.

> What are you planning to work on soon?

* Get Type config PR ready.
* Raise discussions for APIs for Database and Display options in type selection menu.
* Review and merge current open frontend PRs.
* Create issues required for data types milestone.
* Prepare for CodeForJapan hackathon.

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

None

**Comment by ghislaineguerin** [comment]
### Ghislaine's check-in

> Any thoughts on the concept/format of the new Views product spec?

I discussed some of it w @kgodey. I will add more thoughts to the discussion. It makes sense that we are revising the spec as design exploration uncovered similar issues w the original one.

> How did your most recent work go?

Although very focused on research and getting some prototype changes in for implementing design for Views, it went well.

> What are you planning to work on soon?

Update spec process section on wiki mathesar-wiki#18
Merge Left Sidebar Improvements mathesar-wiki#19
Merge Split Date/Time into Date, Time, and DateTime types mathesar-wiki#21
Exploratory work for Views in the design prototype

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

None

**Comment by silentninja** [comment]
> Any thoughts on the concept/format of the new Views product spec?

I have been busy with the error handling PR and haven't fully reviewed it yet, but it looks good and well thought of

> How did your most recent work go?

I was able to finish off error handling for most cases and get some reviews and discussions done.

> What are you planning to work on soon?

Focus on splitting the error handling and new error structure commits into individual PR. 
Work on Column name ->id Issue once work on the error handling PR is complete
Prepare for CodeForJapan hackathon.

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

None

**Comment by seancolsen** [comment]
## Sean's check in

> Any thoughts on the concept/format of the new Views product spec?

- I haven't had a chance to look at it yet. I'll provide meta-feedback either here or within the discussion if I have any.

> How did your most recent work go?

- Last week felt a bit less productive than normal. I'd say that's partly due to a higher than normal amount of context switching for discussions and small tasks, and partly due to continued distractions from home with our kids still out of school because of the covid surge here.

> What are you planning to work on soon?

- I'll be continuing to highly prioritize addressing review feedback to my PRs and keeping them free of conflicts so that Pavish can merge them.
- Open a PR that demonstrates using multiple modals, layered on top of each other. Then use that PR as the basis for a discussion with Pavish and Ghislaine about how to approach the UX for adding constraints. If this all goes smoothly, we should be able to add table constraints by the end of the week. If we need to re-think the UX here, then it could take longer.
- Pavish and I are planning a sync call for today or tomorrow.
- Then I'll dig into `[05.3] 2022-01 improvements`

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

- It will be great if Pavish can get my three substantive PRs merged this week. I know he's working on this, and it seems like we're close.

**Comment by kgodey** [comment]
> Any thoughts on the concept/format of the new Views product spec?

I like the idea of product specs, it keeps all the information related to a feature in one place and it's easier to think about the feature as a whole (hopefully also easier to review). That way, people also don't have to review individual GitHub issues.

> How did your most recent work go?

It went well, it's nice to make some progress on figuring out Views.

> What are you planning to work on soon?

- Get the Views spec done and merged
- Update remaining Views design tickets to match the new spec
- Think through representing filtering based on CTEs to users (and when that would make sense in our roadmap), see:
    1. #846 
    2. #853 
- Make implementation issues for the following design specs if time permits:
    - https://github.com/centerofci/mathesar-wiki/pull/16
    - https://github.com/centerofci/mathesar-wiki/pull/19

> Do you anticipate any dependencies or blockers that someone else on the team can help you with?

It would be great to get a speedy review of the product spec from the remaining reviewers, if I need to do any rethinking of it, it could affect a lot of other work. @dmos62 @ghislaineguerin @mathemancer @pavish @silentninja.
